mlir/lib/Conversion/MemRefToEmitC/MemRefToEmitC.cpp

```diff
@@ -95,7 +97,13 @@ struct ConvertGlobal final : public OpConversionPattern<memref::GlobalOp> {
           op.getLoc(), "global variable with alignment requirement is "
                        "currently not supported");
     }
-    auto resultTy = getTypeConverter()->convertType(op.getType());
+
+    Type resultTy;
+    if (type.getRank() == 0)
+      resultTy = getTypeConverter()->convertType(type.getElementType());
```

Comment:
You should be able to just do

resultTy = getTypeConverter()->convertType(getElementTypeOrSelf(type));